avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -933,13 +939,41 @@ private static Object maybeWrapSchemaless(Schema schema, Object value, String ty
         }
       }
 
+      // the new and correct way to handle logical types
+      if (schema.name() != null) {
+        if (Decimal.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
+          String precisionString = schema.parameters().get(CONNECT_AVRO_DECIMAL_PRECISION_PROP);
+          // preserve decimal logical type iff precision is specified,
+          // as assuming fixed precision for arbitrary precision record might incur data loss
+          if (precisionString != null) {
+            int precision = Integer.parseInt(precisionString);
+            int scale = Integer.parseInt(schema.parameters().get(Decimal.SCALE_FIELD));
+            org.apache.avro.LogicalTypes.decimal(precision, scale).addToSchema(baseSchema);
+          }
+        } else if (Time.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
+          org.apache.avro.LogicalTypes.timeMillis().addToSchema(baseSchema);
+        } else if (Timestamp.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
+          org.apache.avro.LogicalTypes.timestampMillis().addToSchema(baseSchema);
+        } else if (Date.LOGICAL_NAME.equalsIgnoreCase(schema.name())) {
+          org.apache.avro.LogicalTypes.date().addToSchema(baseSchema);
+        }
+      }
+
+      // Initially, to add support for logical types a new property was added
+      // with key `logicalType`. This enabled logical types for avro schemas but not others,
+      // such as parquet. The use of 'addToSchema` above supersedes this method here,
+      //  which should eventually be removed.
+      // Keeping for backwards compatibility until a major version upgrade happens.
+
+      // OLD COMMENT
```

Comment:
Let's not use all caps in comments, because it's not clear what that means. 
Suggestion: 
```
// Below follows the older method of supporting logical types via properties. It is retained for now and will be deprecated eventually. 
```